src/Response/ErrorResponse.php

```diff
@@ -36,6 +36,7 @@ public static function fromArray(array $response): self
         Assert::keyExists($response, 'error');
         Assert::keyExists($response, 'key');
         Assert::keyExists($response, 'status');
+        Assert::integer($response['status']);
```

Comment:
if you require integer that you don't need to cast into int. Better check will be `::numeric`